src/main/java/com/edmunds/tools/databricks/maven/BaseDatabricksJobMojo.java

```diff
@@ -71,264 +71,160 @@
-
-    protected JobService getJobService() {
-        return getDatabricksServiceFactory().getJobService();
-    }
-
-    /**
-     * NOTE - only for unit testing!
-     */
-    void setFailOnDuplicateJobName(boolean failOnDuplicateJobName) {
-        this.failOnDuplicateJobName = failOnDuplicateJobName;
-    }
-
-    /**
-     * NOTE - only for unit testing!
-     */
-    void setDbJobFile(File dbJobFile) {
-        this.dbJobFile = dbJobFile;
-    }
-
-
-    /**
-     * Check the value of targetDTO and fill targetDTO with defaultDTO if value do not exist.
-     *
-     * @param targetDTO
-     * @param defaultDTO
-     * @param jobTemplateModel
-     */
-    public void fillInDefaultJobSettings(JobSettingsDTO targetDTO, JobSettingsDTO defaultDTO, JobTemplateModel jobTemplateModel) throws JsonProcessingException {
-
-        String jobName = targetDTO.getName();
-        if (StringUtils.isEmpty(targetDTO.getName())) {
-            jobName = jobTemplateModel.getGroupWithoutCompany() + "/" + jobTemplateModel.getArtifactId();
-            targetDTO.setName(jobName);
-            getLog().info(String.format("set JobName with %s", jobName));
-        }
-
-        // email_notifications
-        if (targetDTO.getEmailNotifications() == null) {
-            targetDTO.setEmailNotifications(SerializationUtils.clone(defaultDTO.getEmailNotifications()));
-            getLog().info(String.format("%s|set email_notifications with %s"
-                    , jobName, OBJECT_MAPPER.writeValueAsString(defaultDTO.getEmailNotifications())));
-
-        } else if (targetDTO.getEmailNotifications().getOnFailure() == null
-                || targetDTO.getEmailNotifications().getOnFailure().length == 0
-                || StringUtils.isEmpty(targetDTO.getEmailNotifications().getOnFailure()[0])) {
-            targetDTO.getEmailNotifications().setOnFailure(defaultDTO.getEmailNotifications().getOnFailure());
-            getLog().info(String.format("%s|set email_notifications.on_failure with %s"
-                    , jobName, OBJECT_MAPPER.writeValueAsString(defaultDTO.getEmailNotifications().getOnFailure())));
-        }
-
-        // ClusterInfo
-        if (StringUtils.isEmpty(targetDTO.getExistingClusterId())) {
-            if (targetDTO.getNewCluster() == null) {
-                targetDTO.setNewCluster(SerializationUtils.clone(defaultDTO.getNewCluster()));
-                getLog().info(String.format("%s|set new_cluster with %s", jobName, OBJECT_MAPPER.writeValueAsString(defaultDTO.getNewCluster())));
-
-            } else {
-                if (StringUtils.isEmpty(targetDTO.getNewCluster().getSparkVersion())) {
-                    targetDTO.getNewCluster().setSparkVersion(defaultDTO.getNewCluster().getSparkVersion());
-                    getLog().info(String.format("%s|set new_cluster.spark_version with %s", jobName, defaultDTO.getNewCluster().getSparkVersion()));
+    private SettingsInitializer<JobTemplateModel, JobSettingsDTO> createSettingsInitializer() {
```

Comment:
Should this be in its own class?

Reply:
Probably yes, it could simplify Mojo class - will try it